### DISCUSSION TITLE
Parrot on Windows 11

### DISCUSSION THREAD

**Discussion by franticflux** [frankroeder/parrot.nvim#53] [Q&A] [answered]
There seems to be some kind of problem with the environment variable "DEBUG_PARROT" on Windows 11. The problem I have is my terminal freezes. I'm using a standard setup and Anthropic.  The culprit is one of the first lines in logger.lua and I changed line 5 from 

`  _debug_enabled = vim.env.DEBUG_PARROT ~= nil, `
 
to

`_debug_enabled = os.getenv("DEBUG_PARROT") ~= nil,
`

And now it all seems to be working fine. Does anyone know what the problem is? (I guess it has something to do with how powershell/Neovim handles the enviorment variables)

Update: The plugin doesn't seem to be using normalized paths / portability at all and for instance the check
```
M.is_chat = function(buf, file_name, chat_dir)
  if not M.starts_with(file_name, chat_dir) then
    return false
  end
```
Will fail miserably on Windows (and most other operating systems not being linux). So guess I have to rewrite or use a WSL...

**Comment by frankroeder** [comment, marked as answer]
Dear @franticflux,

Thank you very much for pointing out this issue. At the moment, `parrot.nvim` cannot fully support Windows, especially regarding the way files are stored and read (resolving paths, etc.).

I can assure you that it works on Unix and Linux systems. I believe that WSL is the way to go for Windows users. Otherwise, I am always open to incorporating changes that also support Windows, but I currently do not have the time to set up such a machine to do it myself.

Thank you for your understanding.

**Reply by franticflux** [reply]
Well, it seems I was mistaken. sorry about that.  Apparently you can mix "\\\" and "/" in the same path. However the real problem was a never ending loop in the function find_git_root (in file_utils.lua). On Windows you never reach the condition "/" since "c:\" is the root. I fixed it and now everything seems to be working just fine on Windows.  At least everything I have used and tested so far :-)
